nomos-da/network/core/src/protocols/replication/behaviour.rs

```diff
@@ -123,16 +151,14 @@ where
             .filter(|(peer_id, _connection_id)| peers.contains(peer_id))
             .collect();
 
-        for (peer_id, connection_ids) in connected_peers {
-            for connection_id in connection_ids.iter() {
-                self.outgoing_events.push_back(SwarmEvent::NotifyHandler {
-                    peer_id: *peer_id,
-                    handler: NotifyHandler::One(*connection_id),
-                    event: Either::Left(BehaviourEventToHandler::OutgoingMessage {
-                        message: message.clone(),
-                    }),
-                })
-            }
+        for (peer_id, connection_id) in connected_peers {
+            self.outgoing_events.push_back(SwarmEvent::NotifyHandler {
+                peer_id: *peer_id,
+                handler: NotifyHandler::One(*connection_id),
+                event: Either::Left(BehaviourEventToHandler::OutgoingMessage {
+                    message: message.clone(),
+                }),
+            })
```

Comment:
Are we sure we would have a single connection id?

Reply:
Yes, this is related to the removal of multiple connections per peer in replication node in  https://github.com/logos-co/nomos-node/pull/1055/files#diff-b7380ed35739ccc415111debb3d69be00b9f1c257963446c33c68fbed9555078L58
Now a peer can have one connection to it only, if new one is created - the old one will be overwritten.